contracts/StablecoinConverter.sol

```diff
@@ -260,31 +279,39 @@ contract StablecoinConverter is EpochTokenLocker {
                     currentPrices[order.buyToken],
                     currentPrices[order.sellToken]
                 );
+                buyVolume = uint128(buyVolume.mul(feeDenominator - 1) / feeDenominator);
                 revertRemainingOrder(owner, orderId, sellVolume);
                 subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
             }
         }
     }
 
     function findPriceIndex(uint16 index, uint16[] memory tokenIdForPrice) private pure returns (uint) {
+        // return fee token
+        if (index == 0) {
+            return 0;
+        }
+        // binary search for the other tokens
         uint leftValue = 0;
         uint rightValue = tokenIdForPrice.length - 1;
         while (rightValue >= leftValue) {
             uint middleValue = leftValue + (rightValue-leftValue) / 2;
             if (tokenIdForPrice[middleValue] == index) {
-                return middleValue;
+                return middleValue + 1;
             } else if (tokenIdForPrice[middleValue] < index) {
                 leftValue = middleValue + 1;
             } else {
+                require(middleValue > 0, "Price not provided for token, underflow would have happened");
                 rightValue = middleValue - 1;
             }
         }
         revert("Price not provided for token");
     }
 
-    function checkPriceOrdering(uint16[] memory tokenIdForPrice) private pure returns (bool) {
-        for (uint i = 1; i < tokenIdForPrice.length; i++) {
-            if (tokenIdForPrice[i] <= tokenIdForPrice[i - 1]) {
+    function checkPriceOrdering(uint16[] memory tokenIdsForPrice) private pure returns (bool) {
+        require(tokenIdsForPrice[0] > 0, "price for fee token should not be overwritten");
```

Comment:
I don't understand this require. Shouldn't it say something like "fee token price index does not have to be specified"?

Reply:
Yeah, your wording is better.
But the effect of the specification `tokenIdsForPrice[0]` would be that the fee token price is indeed overwritten:

`function updateCurrentPrices(
        uint128[] memory prices,  //list of prices for touched token only, frist price is fee Token price
        uint16[] memory tokenIdsForPrice  // price[i] is the price for the token with tokenID tokenIdsForPrice[i]
    ) internal {
        for (uint i = 0; i < previousSolution.tokenIdsForPrice.length; i++) {
            currentPrices[previousSolution.tokenIdsForPrice[i]] = 0;
        }
        currentPrices[0] = prices[0];
        for (uint i = 0; i < tokenIdsForPrice.length; i++) {
            currentPrices[tokenIdsForPrice[i]] = prices[i + 1];
        }
    }`